grid_map_loader/src/GridMapLoader.cpp

```diff
@@ -22,9 +22,16 @@ GridMapLoader::GridMapLoader()
 {
   readParameters();
 
-  rclcpp::QoS custom_qos(rclcpp::KeepLast(1));  // Buffer only last published message.
-  custom_qos.transient_local();  // Persist messages for “late-joining” subscriptions.
-  custom_qos.lifespan(duration_);  // Expiration duration of persisted messasges.
+  rclcpp::QoS custom_qos(10);  // initialize to default
+
+  if (durationInSec > 0) {
+    // Expiration duration of persisted messasges.
+    custom_qos.lifespan(rclcpp::Duration::from_seconds(durationInSec));
```

Comment:
I'm curious why you're setting lifespan, I don't typically see this QoS being messed with

Reply:
In the original ros1 code:
-The publisher [latch flag](https://github.com/ANYbotics/grid_map/blob/07c8429b7bcabe182b9f2c203b2f10d3de583f3c/grid_map_loader/src/GridMapLoader.cpp#L21) is set to true.
-The [sleep](https://github.com/ANYbotics/grid_map/blob/c796c48d11fb36815a4366123089a2b080735292/grid_map_loader/src/GridMapLoader.cpp#L53) is used to keep the node running for a duration (instead of instant shutdown) to allow nodes that are created after the grid_map_loader node to still receive the published gridMap message (this was verified with @maximilianwulf on slack).

The lifespan is set to the `duration` parameter value along with setting the durability to `Transient local` to ensure that the message is available for the duration requested ` for late joining nodes.

Reply:
If you dont set the lifespan though the new nodes can get an arbitrarily old message as long as its the latest one.

Reply:
Good or bad, not sure, it depends on the lifespan (he-he-he) of the grid map loader object and the context that its used in. For the nav2 map server for instance, I wouldn't want to set this but I'm not sure about this since I don't know off hand how its used.

Reply:
Shall we open an issue or add it to the existing ros2 issue as something to test once we have grid_map_demo ported and we have a gridmap rosbag to play with?

Reply:
The only use of this i see is here: https://github.com/ANYbotics/grid_map/blob/0f673c3600dae5ecf362d75e2e37e1617ff849c2/grid_map_loader/src/grid_map_loader_node.cpp

Should we be making the latched topic only temporary? It doesn't seem to me like that's the right option, this seems to act more like a map server than a time-out-valid temporary costmap

Reply:
> Should we be making the latched topic only temporary?

What do you mean by temporary?

Are you suggesting changing this to be equivalent to map_server i.e. latching topic and always running and publishing to new subscribers?

Reply:
@SteveMacenski Please see my question above, I think we both commented seconds away form each other :')

Reply:
> What do you mean by temporary?

Setting lifespan sets a lifespan for the message so it becomes no longer valid after it expires, so its only latched for that duration. https://design.ros2.org/articles/qos_deadline_liveliness_lifespan.html. So I think you don't want to set lifespan at all and just have it be latched and the duration only be the sleeo.

Reply:
Sure will remove it then. 

The [default lifespan](https://github.com/ros2/rmw/blob/163f0700d0df30b58c95a300d255f834a3805b8c/rmw/include/rmw/types.h#L430-L431) is set to 0, so I was just worried the message might expire spontaneously.

Reply:
We dont set the lifespan at all in nav2 and I think the latching is working with the map server OK. We only make use of the basic 4 QoS where lifespan is part of the RT QoS